server/rpc.go

```diff
@@ -316,15 +322,18 @@ func verifyOrchestratorReq(orch Orchestrator, addr ethcommon.Address, sig []byte
 	return orch.CheckCapacity("")
 }
 
-type authBroadcasterResponse struct{}
+type discoveryAuthWebhookRes struct {
+	PriceInfo *net.PriceInfo `json:"price,omitempty"`
+}
 
 // authenticateBroadcaster returns an error if authentication fails
-func authenticateBroadcaster(id string) (*authBroadcasterResponse, error) {
+// on success it caches the webhook response
+func authenticateBroadcaster(addr ethcommon.Address) (*discoveryAuthWebhookRes, error) {
```

Comment:
I don't think this function needs to know that the ID is an ETH address - the ID can be opaque here leaving its interpretation to the webhook server implementation. Might also be useful to keep the ID as a string if the format is updated later on.

Reply:
fixed in 7dd34f5